test/python/tools/test_compiler.py

```diff
@@ -256,8 +256,7 @@ def test_compile_coupling_map(self):
         qc.measure(qr[2], cr[2])
         shots = 2048
         coupling_map = [[0, 1], [1, 2]]
-        initial_layout = {("qr", 0): ("q", 0), ("qr", 1): ("q", 1),
-                          ("qr", 2): ("q", 2)}
+        initial_layout = {(qr, 0): 0, (qr, 1): 1, (qr, 2): 2}
```

Comment:
Can we keep the old way of passing `initial_layout` working, with a DeprecationWarning?
The old way was that you specified register names, not objects. Just for backward compatibility purposes since this is touching common user-level functions.

Reply:
done in a7b30a02